### ISSUE TITLE
CA1508 false positive in switch expression default case

### ISSUE THREAD

**Issue by AnthonyMastrean** [dotnet/roslyn-analyzers#7337]
### Analyzer

**Diagnostic ID**: [CA1508](https://learn.microsoft.com/en-us/dotnet/fundamentals/code-analysis/quality-rules/ca1508): `Avoid dead conditional code`

### Analyzer source

**SDK**: [Built-in CA analyzers in .NET 5 SDK or later](https://learn.microsoft.com/dotnet/fundamentals/productivity/code-analysis)

**Version**: [SDK 5.0.100](https://dotnet.microsoft.com/download/dotnet/5.0)

### Describe the bug

The analyzer warns on the below code despite no condition that would indicate `value` is not null.

### Steps To Reproduce

```csharp
static class Ca1508FalsePositiveInSwitchExpressionDefaultCase
{
    static void Example(string? value)
    {
        _ = value switch
        {
            _ => value?.Trim(),
        };
    }
}
```

### Expected behavior

No warning!

### Actual behavior

Warning!

```
'value' is never 'null'. Remove or refactor the condition(s) to avoid dead code.
```
